packages/payment-proxy-client/src/file.ts

```diff
@@ -1,29 +1,158 @@
 import { NitroRpcClient } from "@statechannels/nitro-rpc-client";
+import { Voucher } from "@statechannels/nitro-rpc-client/src/types";
 
 export async function fetchFile(
-  baseUrl: string,
-  costPerByte: number,
-  fileSize: number,
+  url: string,
+  paymentAmount: number,
   selectedChannel: string,
   nitroClient: NitroRpcClient
 ): Promise<File> {
   const voucher = await nitroClient.CreateVoucher(
     selectedChannel,
-    fileSize * costPerByte
+    paymentAmount
   );
 
-  const response = await fetch(
-    `${baseUrl}?channelId=${voucher.ChannelId}&amount=${voucher.Amount}&signature=${voucher.Signature}`
+  const response = await fetch(addVoucherToUrl(url, voucher));
+
+  const fileName = parseFileNameFromUrl(response.url);
+
+  return new File([await response.blob()], fileName);
+}
+export async function fetchFileInChunks(
+  chunkSize: number,
+  url: string,
+  costPerByte: number,
+  selectedChannel: string,
+  nitroClient: NitroRpcClient,
+  updateProgress: (progress: number) => void
+): Promise<File> {
+  updateProgress(0);
+
+  const firstChunk = await fetchChunk(
+    0,
+    chunkSize - 1,
+    url,
+    costPerByte,
+    selectedChannel,
+    nitroClient
   );
 
-  try {
-    const fileName = parseFileNameFromUrl(response.url);
-    console.log(fileName);
-    return new File([await response.blob()], fileName);
-  } catch (e) {
-    console.log(e);
-    throw e;
+  const { contentLength, fileName } = firstChunk;
+  let remainingContentLength = contentLength - chunkSize;
+
+  console.log(
+    `Fetched the first chunk of the file using a chunk size of ${chunkSize} bytes`
+  );
+
+  console.log(`The file ${fileName} is ${contentLength} bytes in size`);
+
+  const fileContents = new Uint8Array(contentLength);
```

Comment:
This is probably fine for demo purposes where files are small. However with massive files we probably would want to avoid storing the whole file in memory before writing it to disk.